lib/vdc/BOVCollection.cpp

```diff
@@ -245,38 +288,67 @@ void BOVCollection::_populateDataFileMap()
     _dataFileMap[_variable][_time] = _dataFile;
 }
 
-int BOVCollection::_missingValueError(std::string token) const
+int BOVCollection::_cannotStatFileError() const
+{
+    SetErrMsg(("Unable to get file status for " + _dataFile).c_str());
+    return -1;
+}
+
+int BOVCollection::_invalidFileSizeError(size_t dataSize, size_t fileSize) const
```

Comment:
Again, this seems like a lot of additional code to simply report a file read error. E.g.

```
if ((rc = fread()) < 0) {
    SetErrMsg("Error readfile file %s : %M", path.c_str());
    return(-1);
}
```

Reply:
I see what you mean.

Most (not all) of my error messages here are called through functions.  I think this makes [the validation code](https://github.com/NCAR/VAPOR/blob/a4f4924ad15112a439c27429b1911faa96b1f904/lib/vdc/BOVCollection.cpp#L198) easier to read, even though there are technically more total lines of code in the file.

Currently, the validation for each token is ~8 lines of code and some tests have more than one error condition.  I thought it would be more readable to call something like _inconsitentValueError(), rather than SetErrMsg(message) and then return -1 for [validating the data format across BOV files](https://github.com/NCAR/VAPOR/blob/a4f4924ad15112a439c27429b1911faa96b1f904/lib/vdc/BOVCollection.cpp#L228).  I think it's just slightly more readable this way.

If this doesn't seem worth it, let me know and I can revert the error reporting functions.

Reply:
Just a note to @clyne : `fread()` returns a value of type `size_t` so `if ((rc = fread()) < 0) `  will give you unexpected answers. (Use a static analyzer to catch this type of errors.)

Reply:
My bad. Correct. The return value should be compared with the total number of bytes requested and if they disagree there was an error or a short file.

Reply:
Well, be careful again: compare to the number of objects to read, ***not*** the number of bytes. I.e. if you want to read 10 floats, compare with 10, instead of 40.

Reply:
Indeed. It's the number of elements not bytes

Reply:
Fixed at the end of the _validateParsedValues function.